packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,687 @@
+
+We chose the `Material-UI` theme and our initial code is follows:
+
+*index.js:*
+```jsx
+import {
+  Grid,
+  Table,
+  TableHeaderRow,
+  Toolbar
+} from '@devexpress/dx-react-grid-material-ui';
+import service from './data.js';
+
+class App extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      columns: [
+        { name: "ID", title: "ID" },
+        { name: "CompanyName", title: "Company Name" },
+        { name: "Address", title: "Address" },
+        { name: "City", title: "City" },
+        { name: "State", title: "State" },
+        { name: "Zipcode", title: "Zip code" }
+      ],
+      rows: service.getCustomers()
+    };
+  }
+  render() {
+    const { rows, columns } = this.state;
+    return (
+      <Paper>
+        <Grid
+          rows={rows}
+          columns={columns}
+        >
+          <Table />
+          <TableHeaderRow />
+          <Toolbar />
+        </Grid>
+      </Paper>
+    );
+  }
+}
+```
+*data.js:*
+```js
+const customers = [{
+    ID: 1,
+    CompanyName: 'Super Mart of the West',
+    Address: '702 SW 8th Street',
+    City: 'Bentonville',
+    State: 'Arkansas',
+    Zipcode: 72716
+  }, {
+    ID: 2,
+    CompanyName: 'Electronics Depot',
+    Address: '2455 Paces Ferry Road NW',
+    City: 'Atlanta',
+    State: 'Georgia',
+    Zipcode: 30339
+  }, {
+    ID: 3,
+    CompanyName: 'K&S Music',
+    Address: '1000 Nicllet Mall',
+    City: 'Minneapolis',
+    State: 'Minnesota',
+    Zipcode: 55403
+  }, {
+    ID: 4,
+    CompanyName: "Tom's Club",
+    Address: '999 Lake Drive',
+    City: 'Issaquah',
+    State: 'Washington',
+    Zipcode: 98027
+  }, {
+    ID: 5,
+    CompanyName: 'E-Mart',
+    Address: '3333 Beverly Rd',
+    City: 'Hoffman Estates',
+    State: 'Illinois',
+    Zipcode: 60179
+  }];
+
+export default {
+  getCustomers() {
+    return customers;
+  }
+}
+```
+
+Let's take a look at the screenshot below to see the final result that we will get after performing all the steps from the tutorial:
+
+![](../../../../../img/guides/custom-plugin-development/result.png)
+
+As you can see, there is a select box at the top right corner of the grid associated with the `CompanyName` column. When a user selects an item from the box, the grid displays only filtered items. Moreover, there is the `CLEAR` button that clears the selected value. When the value is cleared, the grid displays all available rows.
+
+
+Let's start extending our grid!
+
+## Rendering Custom Markup
+For rendering custom markup in existing grid UI elements, in our case it is the toolbar, we need to create a custom plugin. For creating a plugin, we need to wrap all content in the [Plugin](../../../core/docs/reference/plugin.md) component. Since our plugin will reside in the grid's toolbar, we will name it `ToolbarFilter`. Let's create a new folder `plugins` in our project and
```

Comment:
I think 'To create a plugin' sounds better considering the previous sentence with same construction

Reply:
I'm not sure I understand in what part of the text you wish to use 'To create plugin'. Would you please clarify?

Reply:
Instead of 'For creating a plugin'

Reply:
Ok, thanks